eng/common/scripts/Helpers/git-helpers.ps1

```diff
@@ -110,3 +110,143 @@ class ConflictedFile {
+#       "reset": 1722875519
+#     },
+#     "graphql": {
+#       "limit": 5000,
+#       "used": 0,
+#       "remaining": 5000,
+#       "reset": 1722879059
+#     },
+#     "integration_manifest": {
+#       "limit": 5000,
+#       "used": 0,
+#       "remaining": 5000,
+#       "reset": 1722879059
+#     },
+#     "source_import": {
+#       "limit": 100,
+#       "used": 0,
+#       "remaining": 100,
+#       "reset": 1722875519
+#     },
+#     "code_scanning_upload": {
+#       "limit": 1000,
+#       "used": 0,
+#       "remaining": 1000,
+#       "reset": 1722879059
+#     },
+#     "actions_runner_registration": {
+#       "limit": 10000,
+#       "used": 0,
+#       "remaining": 10000,
+#       "reset": 1722879059
+#     },
+#     "scim": {
+#       "limit": 15000,
+#       "used": 0,
+#       "remaining": 15000,
+#       "reset": 1722879059
+#     },
+#     "dependency_snapshots": {
+#       "limit": 100,
+#       "used": 0,
+#       "remaining": 100,
+#       "reset": 1722875519
+#     },
+#     "audit_log": {
+#       "limit": 1750,
+#       "used": 0,
+#       "remaining": 1750,
+#       "reset": 1722879059
+#     },
+#     "audit_log_streaming": {
+#       "limit": 15,
+#       "used": 0,
+#       "remaining": 15,
+#       "reset": 1722879059
+#     },
+#     "code_search": {
+#       "limit": 10,
+#       "used": 0,
+#       "remaining": 10,
+#       "reset": 1722875519
+#     }
+#   },
+#   "rate": {
+#     "limit": 5000,
+#     "used": 1087,
+#     "remaining": 3913,
+#     "reset": 1722876411
+#   }
+# }
+
+# These are the rate limit types we care about. If others needed in the future they
+# can be defined here. The reason these need to be defined is because Get-RateLimit
+# call needs to select the particular property to return the right limit. This ensures
+# that rate limit type being passed to the function will exist.
+enum RateLimitTypes {
+  core
+  search
+  graphql
+}
+
+# Fetch the rate limit for the given RateLimitType
+# TODO: Possibly change the parameter to take in an array of types and return an
+# array of limits?
+function Get-RateLimit([RateLimitTypes]$RateLimitType) {
+  $returnValue = gh api rate_limit
+  if ($LASTEXITCODE) {
+    LogError "Get-RateLimit::unable to get rate limit"
+    exit $LASTEXITCODE
+  }
+  # All rate limits have the following fields: limit, used, remaning, reset.
+  # Returning -AsHashtable allows easier access, eg. $rate_limit.remaining
+  $rate_limit = $returnValue | ConvertFrom-Json -AsHashtable | Select-Object -ExpandProperty resources | Select-Object -ExpandProperty $RateLimitType
+  # Add the limit type for convenance
+  $rate_limit["type"] = $RateLimitType
+  return $rate_limit
+}
+
+# Get the number of minutes until RateLimit reset rounded up to the nearest minute
+# for the passed in RateLimit. This is more applicable to the core and graphql rate
+# limits than search because the search rate limit resets every minute
+function Get-MinutesUntilRateLimitReset($RateLimit) {
```

Comment:
I don't see this used anywhwere.

Reply:
No, because it isn't yet. Recall that these changes are initially going to be gathering data, until we know how many calls we're using or whether or not we hit rate limits, I can't really use this to wait until rate limit reset.